executor/builder.go

```diff
@@ -601,6 +601,16 @@ func (b *executorBuilder) buildSelectLock(v *plannercore.PhysicalLock) Executor
 		tblID2Handle:     v.TblID2Handle,
 		partitionedTable: v.PartitionedTable,
 	}
+	if len(e.partitionedTable) > 0 {
+		schema := v.Schema()
+		e.tblID2PIDColumnIndex = make(map[int64]int)
+		for i := 0; i < len(v.ExtraPIDInfo.Columns); i++ {
+			col := v.ExtraPIDInfo.Columns[i]
+			tblID := v.ExtraPIDInfo.TblIDs[i]
+			offset := schema.ColumnIndex(col)
```

Comment:
Check the column.ID is better

Reply:
Check the column to be `ExtraPidColID` can get the column index in the schema, but can't build the mapping of 

table ID => partition column index in the schema

Here we take each partition column (v.ExtraPIDInfo.Columns[i]) and find its index in the schema (offset), to build the
mapping `tblID(v.ExtraPIDInfo.TblIDs[i])` => `offset`